### PR TITLE
Use conda forge to build package

### LINKED ISSUE #189: Use conda forge to build package
#### Description
For details see https://github.com/KhiopsML/pilotage/issues/13

#### Questions/Ideas
- Eliminate the `default` channel from the builds and tests
- Use only `conda-forge` and `khiops-dev`/`khiops` for build and tests

### PR DESCRIPTION
**Pull request by popescu-v**
closes #189

### CODE REVIEW

File: .github/workflows/conda.yml [hunk 1]
```diff
@@ -39,19 +39,18 @@ jobs:
           # Checkout the full repository to have the tags so versioneer works properly
           # See issue https://github.com/actions/checkout/issues/701
           fetch-depth: 0
-      - name: Install Miniconda
+      - name: Install Miniforge3
```
Comment (folmos-at-orange) [inline .github/workflows/conda.yml:42]: Just `Miniforge` (without `3`) and similarly all below

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -63,11 +63,11 @@ jobs:
...
           # Native Khiops-based Conda environment
-          /root/miniconda3/bin/conda install -y -n py${{ matrix.python-version }} -c conda-forge unittest-xml-reporting
-          /root/miniconda3/bin/conda install -y -n py${{ matrix.python-version }} --file test-requirements.txt
+          /root/miniforge3/bin/conda install -y -n py${{ matrix.python-version }} unittest-xml-reporting
+          /root/miniforge3/bin/conda install -y -n py${{ matrix.python-version }} --file test-requirements.txt
           # `khiops-core`-based Conda environment
-          /root/miniconda3/bin/conda install -y -n py${{ matrix.python-version }}_conda -c conda-forge unittest-xml-reporting
-          /root/miniconda3/bin/conda install -y -n py${{ matrix.python-version }}_conda --file test-requirements.txt
+          /root/miniforge3/bin/conda install -y -n py${{ matrix.python-version }}_conda unittest-xml-reporting
+          /root/miniforge3/bin/conda install -y -n py${{ matrix.python-version }}_conda --file test-requirements.txt
```
Comment (folmos-at-orange) [inline .github/workflows/unit-tests.yml:70]: Why use the full path? The documentation of setup-miniconda shows that you can just use `conda`. I'd change it in the CI worflow to improve readability.

**Review by folmos-at-orange — CHANGES_REQUESTED** [review]

File: .github/workflows/conda.yml (hunk 1 above)
Comment (popescu-v) [reply .github/workflows/conda.yml:42]: Ok.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (popescu-v) [reply .github/workflows/unit-tests.yml:70]: It was just the minimal change with respect to what we had before. And it was to emphasize that we use the "Conda" that we installed, in the relevant directory.

But IMHO we can try to simplify this and just use `conda` in the PR that would deal with issue https://github.com/KhiopsML/khiops-python/issues/191.

**Review by folmos-at-orange — APPROVED** [review]

File: .github/workflows/conda.yml (hunk 1 above)
Comment (popescu-v) [reply .github/workflows/conda.yml]: Done.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (popescu-v) [reply .github/workflows/unit-tests.yml]: > Why use the full path? The documentation of setup-miniconda shows that you can just use `conda`. I'd change it in the CI worflow to improve readability.

I have tried; it doesn't work: https://github.com/KhiopsML/khiops-python/actions/runs/10456619218/job/28954341545#step:8:18.

Comment (folmos-at-orange) [reply .github/workflows/unit-tests.yml:70]: Okay.

File: .github/workflows/unit-tests.yml [hunk 3]
```diff
@@ -70,11 +70,11 @@ jobs:
...
```
Comment (folmos-at-orange) [inline .github/workflows/unit-tests.yml:77]: I'd just add two environment variables to improve readability (I think we did a similar thing in khiops core)
```bash
CONDA="/root/miniforge3/bin/conda"
CONDA_ENVS="py${{ matrix.python-version }} py${{ matrix.python-version }}_conda"
for CONDA_ENV in "$CONDA_ENVS"
do
  mkdir -p -m "$CONDA_ENV"
  $CONDA install -y -n "$CONDA_ENV" unittest-xml-reporting
  $CONDA install -y -n "$CONDA_ENV" --file test-requirements.txt
done
```

File: .github/workflows/unit-tests.yml [hunk 4]
```diff
@@ -113,13 +113,13 @@ jobs:
...
+          /root/miniforge3/bin/conda run --no-capture-output -n py${{ matrix.python-version }} coverage xml -o "reports/py${{ matrix.python-version }}/py-coverage.xml"
           # `khiops-core`-based Conda environments
-          /root/miniconda3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage run -m xmlrunner -o "reports/py${{ matrix.python-version }}_conda" -v
-          /root/miniconda3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage report -m
-          /root/miniconda3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage xml -o "reports/py${{ matrix.python-version }}_conda/py-coverage.xml"
+          /root/miniforge3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage run -m xmlrunner -o "reports/py${{ matrix.python-version }}_conda" -v
+          /root/miniforge3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage report -m
+          /root/miniforge3/bin/conda run --no-capture-output -n py${{ matrix.python-version }}_conda coverage xml -o "reports/py${{ matrix.python-version }}_conda/py-coverage.xml"
```
Comment (folmos-at-orange) [inline .github/workflows/unit-tests.yml:122]: Same refactoring as above.

**Review by folmos-at-orange — CHANGES_REQUESTED** [review]

File: .github/workflows/unit-tests.yml (hunk 3 above)
Comment (popescu-v) [reply .github/workflows/unit-tests.yml]: Ok & Done.

File: .github/workflows/unit-tests.yml (hunk 4 above)
Comment (popescu-v) [reply .github/workflows/unit-tests.yml]: Ok & Done.

**Review by folmos-at-orange — APPROVED** [review]

**Pull request merged**